subprojects/scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java

```diff
@@ -156,14 +151,13 @@ private boolean isNonIncrementalCompilation() {
         return false;
     }
 
-    @Nullable
     @Internal
     protected JavaInstallationMetadata getToolchain() {
-        return javaLauncher.map(JavaLauncher::getMetadata).getOrNull();
+        return javaLauncher.map(JavaLauncher::getMetadata).get();
     }
 
     protected ScalaJavaJointCompileSpec createSpec() {
-        validateConfiguration();
+        // TODO: do we need to introduce validations for inconsistent toolchain vs executable/javaHome?
```

Comment:
🤔 This seems like a problem with the current state that the executable/Java home from the fork options isn't taken into account. I think this is a bug which was introduced when we started to support toolchains for Java compilation. 

I am not sure what to do here, maybe keep the current behavior that the executable/javaHome is ignored?

Reply:
Yeah, I also think it does not make sense to add the support for this now.